### PR TITLE
Add Python verson 2.7 and 3.6 to Azure Pipeline

### PR DESCRIPTION
**Pull request by Abhishek-Dev09**
Add python version 2.7 and 3.6 to Azure Pipeline to perform test on both Python 2.7 and Python 3.6 on macOS ,window,Linux.

Signed-off-by: Abhishek Kumar <abhishek.kasyap09@gmail.com>

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+trigger:
+  branches:
+    include: ['*']
+  tags:
+    include: ['*']
+
+jobs:
+  - job: Test
+  pool:
+    vmImage: 'Ubuntu-18.04'
```
Comment (pombredanne) [inline azure-pipelines.yml]: That's OK to test on Ubuntu-18.04 but I would want to have Ubuntu-16.04 too
Which are all the other available VMs? (including mac and windows?)

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    vmImage: 'macOS-10.13'
+    vmImage: 'vs2017-win2016'
+  strategy:
+    matrix:
+      Python36:
+        python.version: '2.7'
+      Python37:
+        python.version: '3.6'
+    maxParallel: 2
```
Comment (pombredanne) [inline azure-pipelines.yml]: why limit the parallelism? Faster is better!

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    
+    displayName: Performing test 
+    steps:
+      - task: UsePythonVersion@0
+      inputs:
+      versionSpec: '$(python.version)'
+      - script: ./configure
```
Comment (pombredanne) [inline azure-pipelines.yml]: Will this syntax also work on Windows?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      - script: ./bin/py.test -vvs tests/commoncode
+      - task: PublishTestResults@2
```
Comment (pombredanne) [inline azure-pipelines.yml]: What is this doing?

**Comment by pombredanne** [comment]
@Abhishek-Dev09 you also need to rebase your changes against the latest develop

File: azure-pipelines.yml (hunk 1 above)
Comment (Abhishek-Dev09) [reply azure-pipelines.yml]: Ok, i will do 16.04.

File: azure-pipelines.yml (hunk 2 above)
Comment (Abhishek-Dev09) [reply azure-pipelines.yml]: Ok, i will remove this.

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    include: ['*']
+  tags:
+    include: ['*']
+
+jobs:
+  - job: Test
+  pool:
+    vmImage: 'Ubuntu-18.04'
+    vmImage: 'macOS-10.13'
+    vmImage: 'vs2017-win2016'
```
Comment (Abhishek-Dev09) [inline azure-pipelines.yml]: Is this line correct?

File: azure-pipelines.yml (hunk 3 above)
Comment (Abhishek-Dev09) [reply azure-pipelines.yml]: Ok, we use simply configure

File: azure-pipelines.yml (hunk 4 above)
Comment (Abhishek-Dev09) [reply azure-pipelines.yml]: Publish the result like this https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/test/publish-code-coverage-results?view=azure-devops

File: azure-pipelines.yml (hunk 5 above)
Comment (Abhishek-Dev09) [reply azure-pipelines.yml]: Or we have add separate blocks for each vm.

**Comment by pombredanne** [comment]
> > @Abhishek-Dev09 you also need to rebase your changes against the latest develop
> 
> Why?

You branch does not merge ... "This branch has conflicts that must be resolved"

File: azure-pipelines.yml (hunk 1 above)
Comment (pombredanne) [reply azure-pipelines.yml]: I would like to keep 18 and add 16. And also possibly any other OS that they may have available

File: azure-pipelines.yml (hunk 5 above)
Comment (pombredanne) [reply azure-pipelines.yml]: No idea. That's for you to figure out

**Comment by pombredanne** [comment]
BTW this is failing as you azure pipeline is not valid YAML

**Comment by pombredanne** [comment]
Please see also https://dev.azure.com/nexB/license-expression/_build?definitionId=2 that is working OK on all OSes https://github.com/nexB/license-expression/

File: azure-pipelines.yml (hunk 1 above)
Comment (pombredanne) [reply azure-pipelines.yml]: thanks. I see you have added 14 and 16.

**Comment by Abhishek-Dev09** [comment]
@pombredanne : Check this code http://dpaste.com/0EHN8EG

**Comment by Abhishek-Dev09** [comment]
`trigger:
  branches:
    include: ['*']
  tags:
    include: ['*']
jobs:
- job: Linux
  pool:
    vmImage: 'ubuntu-16.04'
  strategy:
    matrix:
      ubuntu14:
        containerImage: ubuntu:14.04
      ubuntu16:
       containerImage: ubuntu:16.04
      ubuntu18:
        containerImage: ubuntu:18.04
      Python27:
        python.version: '2.7'
      Python36:
        python.version: '3.6'
  container: $[ variables['containerImage'] ]
  displayName: Performing test 
  steps:
  - task: UsePythonVersion@0
    inputs:
      versionSpec: '$(python.version)'
      architecture: 'x64'
  - script: ./configure
  - script: ./bin/py.test -vvs tests/commoncode
  - task: PublishTestResults@2
    inputs:
      testResultsFiles: '**/test-results.xml'
      testRunTitle: 'Python $(python.version)'
    condition: succeededOrFailed()

- job: macOS
  pool:
    vmImage: 'macOS-10.13'
  strategy:
    matrix:
      macOS10.14:
        containerImage: macOS:10.14
      macOS10.13:
       containerImage: macOS:10.13
      macOS10.12:
        containerImage: macOS:10.12
      Python27:
        python.version: '2.7'
      Python36:
        python.version: '3.6'
  container: $[ variables['containerImage'] ]
  displayName: Performing test 
  steps:
  - task: UsePythonVersion@0
    inputs:
      versionSpec: '$(python.version)'
      architecture: 'x64'
  - script: ./configure
  - script: ./bin/py.test -vvs tests/commoncode
  - task: PublishTestResults@2
    inputs:
      testResultsFiles: '**/test-results.xml'
      testRunTitle: 'Python $(python.version)'
    condition: succeededOrFailed()

- job: Windows
  pool:
    vmImage: 'vs2017-win2016'
  strategy:
    matrix:
      windows2019:
        containerImage: windows-2019
      vs2015-win2012r2:
       containerImage: vs2015-win2012r2
      vs2017-win2016:
        containerImage: vs2017-win2016
      Python27:
        python.version: '2.7'
      Python36:
        python.version: '3.6'
  container: $[ variables['containerImage'] ]
  displayName: Performing test 
  steps:
  - task: UsePythonVersion@0
    inputs:
      versionSpec: '$(python.version)'
      architecture: 'x64'
  - script: configure
  - script: ./bin/py.test -vvs tests/commoncode
  - task: PublishTestResults@2
    inputs:
      testResultsFiles: '**/test-results.xml'
      testRunTitle: 'Python $(python.version)'
    condition: succeededOrFailed()
`

**Comment by Abhishek-Dev09** [comment]
`trigger:
  branches:
    include: ['*']
  tags:
    include: ['*']
jobs:
- job: Linux
  pool:
    vmImage: 'ubuntu-16.04'
  strategy:
    matrix:
      ubuntu14:
        containerImage: ubuntu:14.04
      ubuntu16:
       containerImage: ubuntu:16.04
      ubuntu18:
        containerImage: ubuntu:18.04
      Python27:
        python.version: '2.7'
      Python36:
        python.version: '3.6'
  container: $[ variables['containerImage'] ]
  displayName: Performing test 
  steps:
  - task: UsePythonVersion@0
    inputs:
      versionSpec: '$(python.version)'
      architecture: 'x64'
  - script: ./configure
  - script: ./bin/py.test -vvs tests/commoncode
  - task: PublishTestResults@2
    inputs:
      testResultsFiles: '**/test-results.xml'
      testRunTitle: 'Python $(python.version)'
    condition: succeededOrFailed()

- job: macOS
  pool:
    vmImage: 'macOS-10.13'
  strategy:
    matrix:
      macOS10.14:
        containerImage: macOS:10.14
      macOS10.13:
       containerImage: macOS:10.13
      macOS10.12:
        containerImage: macOS:10.12
      Python27:
        python.version: '2.7'
      Python36:
        python.version: '3.6'
  container: $[ variables['containerImage'] ]
  displayName: Performing test 
  steps:
  - task: UsePythonVersion@0
    inputs:
      versionSpec: '$(python.version)'
      architecture: 'x64'
  - script: ./configure
  - script: ./bin/py.test -vvs tests/commoncode
  - task: PublishTestResults@2
    inputs:
      testResultsFiles: '**/test-results.xml'
      testRunTitle: 'Python $(python.version)'
    condition: succeededOrFailed()

- job: Windows
  pool:
    vmImage: 'vs2017-win2016'
  strategy:
    matrix:
      windows2019:
        containerImage: windows-2019
      vs2015-win2012r2:
       containerImage: vs2015-win2012r2
      vs2017-win2016:
        containerImage: vs2017-win2016
      Python27:
        python.version: '2.7'
      Python36:
        python.version: '3.6'
  container: $[ variables['containerImage'] ]
  displayName: Performing test 
  steps:
  - task: UsePythonVersion@0
    inputs:
      versionSpec: '$(python.version)'
      architecture: 'x64'
  - script: configure
  - script: ./bin/py.test -vvs tests/commoncode
  - task: PublishTestResults@2
    inputs:
      testResultsFiles: '**/test-results.xml'
      testRunTitle: 'Python $(python.version)'
    condition: succeededOrFailed()`

**Pull request closed without merging**